[16.0][ADD] stock_picking_operation_destination_suggestion

rousseldenis opened the pull request.

rousseldenis committed:
[16.0][ADD] stock_picking_operation_destination_suggestion

rousseldenis committed:
[FIX] stock_picking_operation_destination_suggest: Use Char for domain

rousseldenis committed:
[IMP] stock_picking_operation_destination_suggestion: Take incoming operations done

As soon as users filled in the done quantity, take those moves into account to suggest destination

nicolas-delbovier-acsone committed:
[FIX] stock_picking_operation_destination_suggestion: labels disappearing in picking type for view.

It seems odoo UI is only showing a label for a field in its view if the field is inside a LEAF <group> node.
The previous xpath inserted the new "Destination Suggestion" group inside an already existing one making the fields of the later not render their labels.

This commit puts the new "Destination suggestion" group AFTER the xpath-selected group (not inside) and adds a name attribute to ensure easier inheritence.

nicolas-delbovier-acsone commented:
Explanation of my "fix labels disappearing" commit

## Before fix

<img width="1402" height="763" alt="image" src="https://github.com/user-attachments/assets/db4caaa4-af88-4936-9dad-ec56acd998e3" />

---

<img width="1402" height="763" alt="image" src="https://github.com/user-attachments/assets/c2bb8254-38e7-493e-8086-bc4f584f04b9" />

---
<img width="1331" height="393" alt="image" src="https://github.com/user-attachments/assets/9359834f-8123-4026-b258-01e4069733c7" />

## After fix

<img width="1381" height="793" alt="image" src="https://github.com/user-attachments/assets/04afa9af-e5ed-420b-b164-3bc469d67ccb" />

---

<img width="1381" height="793" alt="image" src="https://github.com/user-attachments/assets/8cf0c28a-b84c-46ea-9f4f-82158dfa22e4" />

nicolas-delbovier-acsone commented:
Can we re-open this PR?

lmignon commented on stock_picking_operation_destination_suggestion/models/stock_picking.py:
```diff
@@ -0,0 +1,74 @@
+    )
+    def _compute_destination_location_suggestion_ids(self):
+        for picking in self:
+            pending_out_line_ids = (
+                picking.location_dest_id.children_ids.pending_out_move_line_ids
+            )
+            pending_in_done_line_ids = (
+                picking.location_dest_id.children_ids.pending_in_move_line_ids.filtered(
+                    lambda l: l.qty_done > 0
+                )
+            )
+            domain = self._get_location_destination_move_line_suggestion_domain()
+            locations = (
+                pending_out_line_ids.filtered_domain(domain).location_id
+                | pending_in_done_line_ids.filtered_domain(domain).location_dest_id
+                if picking.picking_type_id.suggest_destination
+                else self.env["stock.location"].browse()
+            )
+            picking.destination_location_suggestion_ids = locations
```
😱 You must filter your pickings before starting to collect the moves used to compute the suggestions...

lmignon committed:
[IMP] stock_picking_operation_destination_suggestion: Improve performance

Do not compute stock move lines for pickings that do not have
suggest_destination enabled.
